### PR TITLE
Build stdlib with per line timings + bench stdlib with -j1

### PR DESCRIPTION
**Pull request by SkySkimmer**
We introduce an option -timef to output the time data to a file instead of having to redirect stdout

Depends 
- https://github.com/coq/coq/pull/17430

### CODE REVIEW

File: tools/dune_rule_gen/coq_module.ml
Comment (SkySkimmer) [inline tools/dune_rule_gen/coq_module.ml:10]: @coq/build-maintainers I don't known if we want to always build with timing or if there should be some way to control it.

File: tools/dune_rule_gen/coq_rules.ml
Comment (SkySkimmer) [inline tools/dune_rule_gen/coq_rules.ml:226]: NB: with display=short, `foo.timing` is nicer than `foo.v.timing`

File: tools/dune_rule_gen/coq_module.ml
Comment (ejgallego) [reply tools/dune_rule_gen/coq_module.ml:10]: I don't know either, it is easy to make it param to the script if needed.

**Comment by SkySkimmer** [comment]
-j 1 is about 2x faster? WTF

**Comment by SkySkimmer** [comment]
The -j 1 job took 6 more min though (17min vs 11min)

**Comment by SkySkimmer** [comment]
Ah
~~~

 Performance counter stats for 'dune build -p coq-stdlib -j 1 @install':

 1,429,296,913,762      instructions:u            #    1.44  insn per cycle         
   990,092,767,616      cycles:u                                                    

     237.111346552 seconds time elapsed

     221.485004000 seconds user
      16.088527000 seconds sys

~~~
vs
~~~
 Performance counter stats for 'dune build -p coq-stdlib -j 12 @install':

 1,429,720,282,762      instructions:u            #    0.83  insn per cycle         
 1,728,730,146,990      cycles:u                                                    

      83.524519649 seconds time elapsed

     402.778996000 seconds user
      30.211645000 seconds sys
~~~

**Comment by SkySkimmer** [comment]
So are we doing anything with this @coq/bench-maintainers @coq/build-maintainers ?

**Comment by SkySkimmer** [comment]
ping @coq/bench-maintainers @coq/build-maintainers

File: tools/dune_rule_gen/coq_rules.ml
Comment (JasonGross) [inline tools/dune_rule_gen/coq_rules.ml:242]: Why are time flags not adjusted?  (And should there be a comment about this?)

**Review by JasonGross — APPROVED** [review]

**Comment by ppedrot** [comment]
I'm willing to merge this but have no knowledge of the dune trickery in it. What's the status of this.

**Comment by Alizter** [comment]
dune stuff lgtm, we can make it nicer later if needed.

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
Merging soon then.

**Comment by ejgallego** [comment]
> I'm willing to merge this but have no knowledge of the dune trickery in it. What's the status of this.

@ppedrot Dune trickery is not so complex. For each .v file, we declare a rule, it looks like:
```lisp
(rule
  (deps foo.v bar.vo ....)
  (targets foo.vo foo.aux)
  (action (run coqc -R ....)))
```
Dune needs to know the targets of a rule because they usually need to be cleaned up if a rule decides not to output that object anymore. This PR is adding a new `foo.timing` file, so Gaëtan updated the code to reflect that in the rule.

**Comment by ppedrot** [comment]
@coqbot merge now

**Comment by SkySkimmer** [comment]
Looks like this broke base+async (missing option filter)

**Comment by SkySkimmer** [comment]
It seems PArith sometimes generates an empty timing file, eg https://gitlab.com/coq/coq/-/jobs/4446472233
~~~
Mismatch between /home/gitlab-runner/builds/av_eR6za/0/coq/coq/_bench/opam.OLD/ocaml-OLD/.opam-switch/build/coq-stdlib.dev/_build/default/theories//./PArith/PArith.timing and /home/gitlab-runner/builds/av_eR6za/0/coq/coq/_bench/opam.NEW/ocaml-NEW/.opam-switch/build/coq-stdlib.dev/_build/default/theories//./PArith/PArith.timing: different measurement counts
~~~
(one is empty, the other has the timings for the single command in the source file)

I have no clue why.

**Comment by ejgallego** [comment]
@SkySkimmer flushing of format buffers can be tricky, and in my experience racy. This is a place to look into IMHO.

**Pull request merged**
